clients/src/main/java/org/apache/kafka/clients/producer/internals/ErrorLoggingCallback.java

```diff
@@ -18,22 +18,41 @@
 
 import org.apache.kafka.clients.producer.Callback;
 import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.header.Headers;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.nio.charset.StandardCharsets;
 
 public class ErrorLoggingCallback implements Callback {
     private static final Logger log = LoggerFactory.getLogger(ErrorLoggingCallback.class);
+    private Headers headers;
     private String topic;
     private byte[] key;
     private byte[] value;
+    private Integer partition;
+    private Long timestamp;
     private int valueLength;
     private boolean logAsString;
 
     public ErrorLoggingCallback(String topic, byte[] key, byte[] value, boolean logAsString) {
+        this(null, topic, key, value, null, null, logAsString);
+    }
+
+    public ErrorLoggingCallback(
+        Headers headers,
+        String topic,
+        byte[] key,
+        byte[] value,
+        Integer partition,
+        Long timestamp,
+        boolean logAsString
+    ) {
```

Comment:
Maybe we should simply pass the `ProducerRecord` in the constructor? We could then also just use the `ProducerRecord.toString` in the error so that we don't have similar issues in the future.